app/src/split/bluetooth/central.c

```diff
@@ -326,6 +351,27 @@ static struct bt_conn_cb conn_callbacks = {
     .disconnected = split_central_disconnected,
 };
 
+int zmk_split_bt_invoke_behavior(const bt_addr_le_t *source, struct zmk_behavior_binding *binding,
+                                 struct zmk_behavior_binding_event event, bool state) {
+    struct zmk_split_run_behavior_payload payload = {.data = {
+                                                         .param1 = binding->param1,
+                                                         .param2 = binding->param2,
+                                                         .position = event.position,
+                                                         .state = state,
+                                                     }};
+    strncpy(payload.behavior_dev, binding->behavior_dev, ZMK_SPLIT_RUN_BEHAVIOR_DEV_LEN - 1);
```

Comment:
It would be nice to verify that this didn't result in truncation. Maybe log an error with the name of the offending behavior?

Also consider using `sizeof(payload.behavior_dev)` instead of `ZMK_SPLIT_RUN_BEHAVIOR_DEV_LEN` so you don't have to change this if you change how the length of `behavior_dev` is determined.

Using strlcpy():
```c
// http://cantrip.org/strlcpy.c
size_t strlcpy(char *dst, const char *src, size_t size)
{
	const size_t len = strlen(src);
	if (size != 0) {
		memcpy(dst, src, (len > size - 1) ? size - 1 : len);
		dst[size - 1] = 0;
	}
	return len;
}

if (strlcpy(payload.behavior_dev, binding->behavior_dev, sizeof(payload.behavior_dev)) >= 
    sizeof(payload.behavior_dev)) {
    DBG_ERR("Truncated!");
}
```
As an addition to this code:
```c
if (strlen(binding->behavior_dev) >= sizeof(payload.behavior_dev)) {
    DBG_ERR("Truncated!");
}
```

Reply:
Fixed! (nearly a year later, oof).